src/main/java/org/opensrp/web/rest/ProductCatalogueResource.java

```diff
@@ -112,6 +129,8 @@
 
 				logger.info("Saving multimedia file...");
 				multimediaService.saveFile(multimediaDTO, file.getBytes(), file.getOriginalFilename());
+				productCatalogue.setPhotoURL(baseUrl + productCatalogue.getUniqueId());
+				productCatalogueService.update(productCatalogue);
```

Comment:
why are we updating again

Reply:
The update method is called again to persist the photo URL in the domain object if there an image provided in the payload.

Reply:
its not possible to update the PK i.e `productCatalogue.getUniqueId()` so on line 99 can we set the photoURL before update so that only 1 update is called. Also you want to check if the image has changed when setting the value so that the value is set when file exists in request

Reply:
Noted.